plugins/palo_alto_cortex_xdr/icon_palo_alto_cortex_xdr/util/api.py

```diff
@@ -41,6 +42,79 @@ def test_connection(self):
                 data=e,
             )
 
+    @Util.retry(tries=1, timeout=900, exceptions=PluginException, backoff_seconds=1)
```

Comment:
This is more a question for myself.
What sorcery is this? Could you leave a comment explaining this line just so I can understand it?
Whether its a comment here in the PR or a comment in the code

Reply:
This uses functools wraps to wrap the current function with another, in this case the retry method in the Util class.

https://docs.python.org/3/library/functools.html

There is a docstring in the retry method of the Util class that explains what's going on there, but essentially it's an exponential backoff.